raiden-ts/tests/unit/raiden.spec.ts

```diff
@@ -418,4 +428,74 @@ describe('Raiden', () => {
     raiden.planUDCWithdraw(withdrawBlock);
     await expect(payload).resolves.toEqual(expect.objectContaining({ block: withdrawBlock }));
   });
+
+  test('withdrawFromUDC', async () => {
+    const withdrawAmount = 10;
+
+    function udcWithdrawEpicMock(action$: Observable<RaidenAction>) {
+      return action$.pipe(
+        filter(udcWithdraw.request.is),
+        mapTo(
+          udcWithdraw.success(
+            {
+              withdrawal: BigNumber.from(withdrawAmount) as UInt<32>,
+              txHash: txHash,
+              txBlock: txBlock,
+              confirmed: true,
+            },
+            {
+              amount: BigNumber.from(100) as UInt<32>,
+            },
+          ),
+        ),
+      );
+    }
+    const deps = makeDummyDependencies();
+    const raiden = new Raiden(
+      makeInitialState(
+        { address, network, contractsInfo },
+        { config: { autoUDCWithdraw: false }, blockNumber: txBlock + 200 },
+      ),
+      deps,
+      combineRaidenEpics(of(initEpicMock, udcWithdrawEpicMock)),
+      dummyReducer,
+    );
+    await expect(raiden.start()).resolves.toBeUndefined();
+
+    const payload = raiden.action$
+      .pipe(
+        first(udcWithdraw.success.is),
+        map((e) => e.payload),
+      )
+      .toPromise();
+
+    raiden.withdrawFromUDC(withdrawAmount);
+    await expect(payload).resolves.toEqual(
+      expect.objectContaining({
+        withdrawal: BigNumber.from(withdrawAmount),
+        txBlock: txBlock,
+        confirmed: true,
+      }),
+    );
+  });
+
+  test('findPFS', async () => {
+    const deps = makeDummyDependencies();
+    const raiden = new Raiden(
+      makeInitialState(
+        { address, network, contractsInfo },
+        { config: { additionalServices: ['pfs1'] }, services: { pfs2: 1 } },
+      ),
+      deps,
+      combineRaidenEpics(of(initEpicMock)),
+      dummyReducer,
+    );
+    await expect(raiden.start()).resolves.toBeUndefined();
+
+    const result = await raiden.findPFS();
```

Comment:
If possible, avoid using `await` outside `expect().resolves` and friends, since if these hang or reject, you're gonna get some uglier tracebacks.

Reply:
✔️